tools/ci_build/github/apple/build_ios_framework.py

```diff
@@ -0,0 +1,149 @@
+            '--ios_sysroot=' + _osx_arch[0],
+            '--osx_arch=' + _osx_arch[1],
+            '--build_dir=' + _build_dir
+        ]
+
+        if args.include_ops_by_config is not None:
+            _build_command += ['--include_ops_by_config=' + args.include_ops_by_config]
+
+        subprocess.run(_build_command, shell=False, check=True, cwd=REPO_DIR)
+
+        _framework_dir = os.path.join(
+            _build_dir, _build_config, _build_config + "-" + _osx_arch[0], 'onnxruntime.framework')
+        _ort_libs.append(os.path.join(_framework_dir, 'onnxruntime'))
+
+        # We actually only need to define the info.plist and headers once since them are all the same
+        if not _info_plist:
+            _info_plist = os.path.join(_build_dir, _build_config, 'info.plist')
+            _headers = glob.glob(os.path.join(_framework_dir, 'Headers', '*.h'))
+
+    # manually create the fat framework
+    _framework_dir = os.path.join(build_dir, 'framework_out', 'onnxruntime.framework')
+    pathlib.Path(_framework_dir).mkdir(parents=True, exist_ok=True)
+
+    # copy the header files and info.plist
+    shutil.copy(_info_plist, _framework_dir)
+    _header_dir = os.path.join(_framework_dir, 'Headers')
+    pathlib.Path(_header_dir).mkdir(parents=True, exist_ok=True)
+    for _header in _headers:
+        shutil.copy(_header, _header_dir)
+
+    # use lipo to create a fat ort library
+    _lipo_command = ['lipo', '-create']
+    _lipo_command += _ort_libs
+    _lipo_command += ['-output', os.path.join(_framework_dir, 'onnxruntime')]
+    subprocess.run(_lipo_command, shell=False, check=True)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        os.path.basename(__file__),
+        description='''Create iOS framework and podspec for one or more osx_archs (fat framework)
+        and building properties specified in the given build config file, see
+        tools/ci_build/github/apple/default_mobile_ios_framework_build_settings.json for details.
+        The output of the final framework and podspec can be found under [build_dir]/framework_out.
+        Please note, this building script will only work on macOS.
+        '''
+    )
+
+    parser.add_argument("--apple_deploy_target", type=str,
+                        help="Specify the minimum version of the target iOS platform")
+
+    parser.add_argument('--build_dir', type=str, default=os.path.join(REPO_DIR, 'build/iOS_framework'),
+                        help='Provide the root directory for build output')
+
+    parser.add_argument(
+        "--include_ops_by_config", type=str,
+        help="Include ops from config file. See /docs/Reduced_Operator_Kernel_build.md for more information.")
+
+    parser.add_argument("--config", type=str, default="Release",
+                        choices=["Debug", "MinSizeRel", "Release", "RelWithDebInfo"],
+                        help="Configuration to build.")
+
+    parser.add_argument('build_settings_file', type=pathlib.Path,
```

Comment:
>'build_settings_file' [](http://example.com/codeflow?start=24&length=21)

seems like this can be used to provide arbitrary build.py parameters. so is it necessary to replicate build.py parameters in the other options as well? if there are required parameters for this particular case, e.g., --apple_deploy_target, we could check that they are present in build_settings_file.

Reply:
can we also move --include_ops_by_config to the build settings file too?
I suppose we can leave --config, as it is slightly handled slightly differently from build.py (expecting a single value here).
perhaps also rename --build_dir to --base_build_dir or --root_build_dir?

Reply:
include_ops_by_config  can point to a file outside of the repo, such that you can build a different package using default setting without any change to the source

Reply:
build_settings_file could also point to a file outside of the repo, right?

Reply:
I guess my point is that having fewer places to specify build.py parameters seems simpler.

Reply:
Moved apple_deploy_target to the build_settings['build_params']